File: front-end/src/app/common/order-item.ts
import {Cart} from "./cart";

export class OrderItem {
  imageUrl: string;
  price: number;
  quantity: number;
  productId: number;

  constructor(cart: Cart) {
    this.imageUrl = cart.imageUrl;
    this.quantity = cart.quantity;
    this.price = cart.price;
    this.productId = cart.id;
  }
}

File: back-end/src/main/java/com/rk/backend/entity/OrderItem.java
package com.rk.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;

    @Column(name="image_url")
    private String imageUrl;

    @Column(name="quantity")
    private int quantity;

    @Column(name="price")
    private double price;

    @Column(name="product_id")
    private int productId;

    @ManyToOne
    @JoinColumn(name="order_id")
    private Order order;
}
